src/Algebra/Graph/AdjacencyIntMap/Algorithm.hs

```diff
@@ -40,65 +40,41 @@ import Data.Tree
 
 import Algebra.Graph.AdjacencyIntMap
 
+import qualified Data.List          as List
 import qualified Data.IntMap.Strict as IntMap
 import qualified Data.IntSet        as IntSet
 
 -- | Compute the /breadth-first search/ forest of a graph, such that
 --   adjacent vertices are explored in increasing order with respect
---   to their 'Ord' instance.
--- 
---   Complexity: /O((n+m)*min(n,W))/ time and /O(n)/ space.
---
--- @
--- bfsForest 'empty'                         == []
--- 'forest' (bfsForest $ 'edge' 1 1)           == 'vertex' 1
--- 'forest' (bfsForest $ 'edge' 1 2)           == 'edge' 1 2
--- 'forest' (bfsForest $ 'edge' 2 1)           == 'vertices' [1,2]
--- 'isSubgraphOf' ('forest' $ bfsForest x) x   == True
--- bfsForest . 'forest' . bfsForest          == bfsForest
--- bfsForest (3 * (1 + 4) * (1 + 5))       == [ Node { rootLabel = 1
---                                                   , subForest = [ Node { rootLabel = 5
---                                                                        , subForest = [] }]}
---                                            , Node { rootLabel = 3
---                                                   , subForest = [ Node { rootLabel = 4
---                                                                        , subForest = [] }]}]
--- 'forest' (bfsForest ('circuit' [1..5] + 'circuit' [5,4..1])) == 'path' [1,2,3] + 'path' [1,5,4]
--- @
-bfsForest :: AdjacencyIntMap -> Forest Int
-bfsForest g = bfsForestFrom' (vertexList g) g
-
--- | Like 'bfsForest', but the traversal is seeded by a list of
---   vertices. Vertices not in the graph are ignored.
+--   to their 'Ord' instance. The search is seeded by a list of
+--   argument vertices that will be the roots of the resulting
+--   forest. Argument vertices not in the graph are ignored.
```

Comment:
I think we should add here that the order of vertices in the list is not important, i.e. we actually treat the list as a set.

Reply:
Oh, wait, I guess I'm wrong, the order does matter!

Reply:
Perhaps, we should have a simple example where the order does matter. Something like `bfs [1,2] (1+2)` will give `[1,2]` whereas `bfs [2,1] (1+2)` will give `[2,1]`, right?

Reply:
In some sense this is subsumed by this example:

```haskell
-- bfsForest vs ('vertices' vs)               == 'map' (\v -> Node v []) ('nub' vs)
```

So, unsure if an extra example is actually needed. I'll let you decide.

Reply:
I think rearranging the order may help make the relations clearer. Probably more information in the documentation as well would help

Reply:
Actually this example could be strengthened:
```
bfsForest g ('vertexList' g)               == 'map' (\v -> Node v []) ('vertexList' g)
```

Reply:
Aha, good example and test case!